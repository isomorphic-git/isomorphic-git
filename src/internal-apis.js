import * as managers from './managers'
import * as models from './models'
import * as utils from './utils'
export { managers }
export { models }
export { utils }
export { listCommits } from './commands/push'
export { listObjects } from './commands/push'
export { pack } from './commands/push'
export { unpack } from './commands/fetch'
