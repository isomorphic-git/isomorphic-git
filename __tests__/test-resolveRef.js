/* global test describe expect */
import _fs from 'fs'
import { models } from '../dist/for-node/internal-apis'
import { resolveRef } from '..'
const { FileSystem } = models
const fs = new FileSystem(_fs)

describe('resolveRef', () => {
  test('1e40fdfba1cf17f3c9f9f3d6b392b1865e5147b9', async () => {
    let ref = await resolveRef({
      fs,
      gitdir: '__tests__/__fixtures__/test-resolveRef.git',
      ref: '1e40fdfba1cf17f3c9f9f3d6b392b1865e5147b9'
    })
    expect(ref).toMatchSnapshot()
  })
  test('test-branch', async () => {
    let ref = await resolveRef({
      fs,
      gitdir: '__tests__/__fixtures__/test-resolveRef.git',
      ref: 'origin/test-branch'
    })
    expect(ref).toMatchSnapshot()
  })
  test('test-tag', async () => {
    let ref = await resolveRef({
      fs,
      gitdir: '__tests__/__fixtures__/test-resolveRef.git',
      ref: 'test-tag'
    })
    expect(ref).toMatchSnapshot()
  })
  test('HEAD', async () => {
    let ref = await resolveRef({
      fs,
      gitdir: '__tests__/__fixtures__/test-resolveRef.git',
      ref: 'HEAD'
    })
    expect(ref).toMatchSnapshot()
  })
  test('HEAD depth', async () => {
    let ref = await resolveRef({
      fs,
      gitdir: '__tests__/__fixtures__/test-resolveRef.git',
      ref: 'HEAD',
      depth: 2
    })
    expect(ref).toMatchSnapshot()
  })
  test('packed-refs', async () => {
    let ref = await resolveRef({
      fs,
      gitdir: '__tests__/__fixtures__/test-resolveRef.git',
      ref: 'v0.0.1'
    })
    expect(ref).toMatchSnapshot()
  })
})
