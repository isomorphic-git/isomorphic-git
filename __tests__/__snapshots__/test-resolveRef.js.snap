// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolveRef 1e40fdfba1cf17f3c9f9f3d6b392b1865e5147b9 1`] = `"1e40fdfba1cf17f3c9f9f3d6b392b1865e5147b9"`;

exports[`resolveRef HEAD 1`] = `"033417ae18b174f078f2f44232cb7a374f4c60ce"`;

exports[`resolveRef HEAD depth 1`] = `"refs/heads/master"`;

exports[`resolveRef packed-refs 1`] = `"1a2149e96a9767b281a8f10fd014835322da2d14"`;

exports[`resolveRef test-branch 1`] = `"e10ebb90d03eaacca84de1af0a59b444232da99e"`;

exports[`resolveRef config 1`] = `"e10ebb90d03eaacca84de1af0a59b444232da99e"`;

exports[`resolveRef test-tag 1`] = `"1e40fdfba1cf17f3c9f9f3d6b392b1865e5147b9"`;
